src/main/java/org/symphonyoss/integration/jira/api/JiraApiResource.java

```
@@ -163,6 +190,58 @@ public ResponseEntity assignIssueToUser(@PathVariable String issueKey,
     }
   }
 
+  /**
+   * Adds new comments on issue.
+   * @param issueKey Issue identifier
+   * @return 201 Created - Returned if add was successful or 400 Bad Request - Returned if the input
+   * is invalid (e.g. missing required fields, invalid values, and so forth), 401 Unauthorized -
+   * Returned if the user is not authenticated, 404 Not Found - Returned if the issue key is not
+   * found.
+   */
+  @PostMapping(value = "/issue/{issueKey}/comment")
+  public ResponseEntity addCommentToAnIssue(@RequestBody(required = false) String comment,
+      @PathVariable String issueKey,
+      @RequestHeader(value = "Authorization", required = false) String authorizationHeader,
+      @RequestParam(name = "url") String jiraIntegrationURL) {
+    Long userId = jwtAuthentication.getUserIdFromAuthorizationHeader(authorizationHeader);
+
+    validateIntegrationBootstrap();
+
+    String accessToken = getAccessToken(jiraIntegrationURL, userId);
+
+    OAuth1Provider provider = getOAuth1Provider(jiraIntegrationURL);
+
+    validateComment(comment);
+
+    JsonNode nodeComment = null;
+    try {
+      nodeComment = JsonUtils.readTree(comment);
+    } catch (IOException e) {
+      malformedCommentExcpetion();
+    }
+
+    String body = nodeComment.path(BODY_PATH).asText();
+    validateComment(body);
```

This validation should be inside the service component. You should throw another exception like 'InvalidJiraCommentException'.